src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -54,5 +104,111 @@ public static ISettings GetSettingsFromDirectory(string directory)
+            {
+                return null;
+            }
+
+            if (showPath)
+            {
+                return item.Value + "\tfile: " + item.ConfigPath;
+            }
+            return item.Value;
+        }
+
+        /// <summary>
+        /// Logs each section of the configuration settings that will be applied, grouped by file path.
+        /// </summary>
+        public static void LogSectionsWithPaths(IEnumerable<string> sections, Settings settings, ILogger logger)
+        {
+            foreach (var section in sections)
+            {
+                logger.LogMinimal(section + ":");
+                var items = settings.GetSection(section).Items;
+
+                IEnumerable<IGrouping<string, SettingItem>> groupByConfigPathsQuery =
+                    from item in items
+                    group item by item.ConfigPath into newItemGroup
+                    select newItemGroup;
+
+                foreach (var configPathsGroup in groupByConfigPathsQuery)
+                {
+                    logger.LogMinimal($" file: {configPathsGroup.Key}");
+                    LogSectionItems(configPathsGroup, logger);
+                    logger.LogMinimal(Environment.NewLine);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Logs each section of the configuration settings that will be applied.
+        /// </summary>
+        public static void LogSectionsNoPaths(IEnumerable<string> sections, Settings settings, ILogger logger)
+        {
+            foreach (var section in sections)
+            {
+                logger.LogMinimal(section + ":");
+                var items = settings.GetSection(section).Items;
+                LogSectionItems(items, logger);
+                logger.LogMinimal(Environment.NewLine);
+            }
+        }
+
+        /// <summary>
+        /// Combines the attributes from each item in a collection of SettingItems into a string, then logs the string.
+        /// </summary>
+        /// <param name="items"></param>
+        /// <param name="logger"></param>
+        public static void LogSectionItems(IEnumerable<SettingItem> items, ILogger logger)
+        {
+            foreach (SettingItem item in items)
+            {
+                var setting = $"\t{item.ElementName}";
+                var attributes = item.GetAttributes();
+                if (attributes != null)
+                {
+                    foreach (var attribute in attributes)
+                    {
+                        setting += $" {attribute.Key}=\"{attribute.Value}\"";
+                    }
+                }
+
+                logger.LogMinimal(setting);
+            }
+        }
+
+        /// <summary>
+        /// Throws an exception if any of the config runner arguments are null.
+        /// </summary>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static void ValidateArguments<TArgs>(TArgs args, Func<ILogger> logger)
```

Comment:
I don't know what's best, but I'd suggest indicating in the name what's being validated. When I see "Validate args", I think of business logic.

```suggestion
        public static void ValidateArgumentsNotNull<TArgs>(TArgs args, Func<ILogger> logger)
```

```suggestion
        public static void EnsureArgumentsNotNull<TArgs>(TArgs args, Func<ILogger> logger)
```